dill/settings.py

```diff
@@ -21,5 +23,177 @@
+    return [x.strip("\"'") for x in option] if strip_quotes else option
+
+def _read_filters(section, module_filters):
+    from logging.config import _resolve
+    from dill.session import EXCLUDE, INCLUDE
+    for rule_type in (EXCLUDE, INCLUDE):
+        rule = rule_type.name.lower()
+        if not any(option.lower().startswith(rule) for option in section):
+            # If no filters, let it fall back to parent module or default filters.
+            delattr(module_filters, rule)
+            continue
+        for option in (rule, '%s.names' % rule, '%s.regexes' % rule):
+            if option in section:
+                for filter in _split_option(section[option], strip_quotes=True):
+                    module_filters.add(filter, rule_type=rule_type)
+        if '%s.types' % rule in section:
+            for name in _split_option(section['%s.types' % rule]):
+                if '.' in name:
+                    filter = _resolve(name)
+                    if not isinstance(filter, type):
+                        raise TypeError("filter is not a type: %r" % filter)
+                else:
+                    filter = 'type:%s' % name
+                module_filters.add(filter, rule_type=rule_type)
+        if '%s.funcs' % rule in section:
+            for code in section['%s.funcs' % rule].strip('\n').splitlines():
+                if code.startswith('(') and code.endswith(')'):
+                    code = code[1:-1]
+                globals_ = {}
+                if not code.startswith('lambda'):
+                    name = code.partition('(')[0]
+                    _resolve(name)  # import required modules
+                    top_level = name.partition('.')[0]
+                    globals_[top_level] = __import__(top_level)
+                filter = eval(code, globals_)
+                if not callable(filter):
+                    raise TypeError("filter is not callable: %r" % filter)
+                module_filters.add(filter, rule_type=rule_type)
+
+def read_settings(filename) -> None:
+    """Read dill settings from an INI file.
+
+    Update the ``dill.settings`` dictionary with the contents of the INI file
+    ``filename``.  Accepted file sections:
+
+      - `dill`: general :py:mod:`dill` settings
+      - `dill.module`: settings for :py:func:`dill.dump_module`
+      - `filters`: default exclude/include filters for :py:func:`dill.dump_module`
+      - `filters.<module_name>`: module-specific filters for
+        :py:func:`dill.dump_module`, where `<module_name>` is the complete module
+        path in the form `module[.submodule...]`
+
+    Accepted option values for general settings:
+
+      - boolean options (case insensitive): yes, no, on, off, true, false
+      - `protocol`: DEFAULT_PROTOCOL, HIGHEST_PROTOCOL, 0, 1, 2, 3, ...
+      - `fmode`: HANDLE_FMODE, 0, CONTENTS_FMODE, 1, FILE_FMODE, 2
+
+    .. IMPORTANT: The demo config file below is used in test_settings.py
```

Comment:
Confusing.  This seems to imply that the config below shows the default settings.  A simple example would be to print the default settings dict, then show a simple config file and read it, and then print the update settings dict.

Reply:
1. The line containing "IMPORTANT" is a comment, doesn't show up in the docs
2. OK, maybe a simpler example is better for the function
3. But I still think that having a complete example file _somewhere_, demonstrating the accepted syntax and features, is important. Could it be in the module's doctstring?

Reply:
Oh, sorry... I forgot that the ReST comments still show up in interactive help

Reply:
either in-file, or module docstring, or sphinx doc with a reference from the source

Reply:
I liked the last option. Hadn't thought of that.